### ISSUE TITLE
TypeError in SRP example

### ISSUE THREAD

**Issue by sanderux** [tomato42/tlslite-ng#130]
after connecting using
```
tls.py client -u alice -p abra123cadabra localhost:4443 
```
(incorrectly syntaxed in README)

Server:
```
tls.py server -v verifierDB localhost:4443
I am an HTTPS test server, I will listen on localhost:4443
Serving files from /home/sander/src/tlslite-ng/tests
Using verifier DB...
About to handshake...
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 36312)
Traceback (most recent call last):
  File "/usr/lib/python2.7/SocketServer.py", line 596, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/local/lib/python2.7/dist-packages/tlslite/integration/tlssocketservermixin.py", line 55, in finish_request
    if self.handshake(tlsConnection) == True:
  File "/usr/local/bin/tls.py", line 345, in handshake
    reqCert=reqCert)
  File "/usr/local/lib/python2.7/dist-packages/tlslite/tlsconnection.py", line 1102, in handshakeServer
    nextProtos=nextProtos, anon=anon):
  File "/usr/local/lib/python2.7/dist-packages/tlslite/tlsconnection.py", line 1131, in handshakeServerAsync
    for result in self._handshakeWrapperAsync(handshaker, checker):
  File "/usr/local/lib/python2.7/dist-packages/tlslite/tlsconnection.py", line 1906, in _handshakeWrapperAsync
    for result in handshaker:
  File "/usr/local/lib/python2.7/dist-packages/tlslite/tlsconnection.py", line 1264, in _handshakeServerAsyncHelper
    settings):
  File "/usr/local/lib/python2.7/dist-packages/tlslite/tlsconnection.py", line 1575, in _serverSRPKeyExchange
    serverKeyExchange = keyExchange.makeServerKeyExchange(sigHash)
  File "/usr/local/lib/python2.7/dist-packages/tlslite/keyexchange.py", line 406, in makeServerKeyExchange
    entry = self.verifierDB[srpUsername]
  File "/usr/local/lib/python2.7/dist-packages/tlslite/basedb.py", line 59, in __getitem__
    valueStr = self.db[username]
  File "/usr/lib/python2.7/bsddb/__init__.py", line 270, in __getitem__
    return _DeadlockWrap(lambda: self.db[key])  # self.db[key]
  File "/usr/lib/python2.7/bsddb/dbutils.py", line 68, in DeadlockWrap
    return function(*_args, **_kwargs)
  File "/usr/lib/python2.7/bsddb/__init__.py", line 270, in <lambda>
    return _DeadlockWrap(lambda: self.db[key])  # self.db[key]
TypeError: String or Integer object expected for key, unicode found
----------------------------------------
```

Tested on Python 2.7.12 and Python 2.7.3

**Comment by tomato42** [comment]
I didn't expect it to be so broken, but I think I have managed to revive it. Could you check the code in #131 ?

@kasserater this may actually be affecting you too

**Comment by sanderux** [comment]
fixed in #131
